mailbox/plugin/spamassassin/src/main/java/org/apache/james/mailbox/spamassassin/SpamAssassin.java

```diff
@@ -41,21 +42,25 @@ public SpamAssassin(MetricFactory metricFactory, SpamAssassinConfiguration spamA
         this.spamAssassinConfiguration = spamAssassinConfiguration;
     }
 
-    public void learnSpam(List<InputStream> messages, Username username) {
+    public Mono<Void> learnSpam(List<MessageToLearn> messages, Username username) {
```

Comment:
there's no batching or anything that would make the code faster with a list AFAICT so we should probably work on a single message at a time.